### PR TITLE
Added the new tool query_impc

### PR DESCRIPTION
**Pull request by AndreaFurlani**
Sorry I made some mistakes with the old branch and I had to create a new one

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by AndreaFurlani** [comment]
> @AndreaFurlani the tests are failing here. Do you need help figuring that out? Also please use macros to reduce the repeating elements in your wrapper.

Hi @bgruening, yes thanks it would be helpful

**Review by bgruening — COMMENTED** [review]
Cool, much better. Now the test needs to be fixed :)

**Comment by AndreaFurlani** [comment]
Hi @bgruening , I'm continuously get the following error during the tests:

[Error_message.txt](https://github.com/galaxyproject/tools-iuc/files/12761588/Error_message.txt)

**Comment by bgruening** [comment]
@AndreaFurlani your local Galaxy instance is too old. As we have added a more recent profile version here.

However, on github the Galaxy version is newer that we use for testing so the error message here in the CI is different.

You can download the tool error reports from this summary page: https://github.com/galaxyproject/tools-iuc/actions/runs/6338840242?pr=5503

Click on All tool tests results:

grafik


If you do that you see three errors:

One "parameter 'g_out': an invalid option ('txt') was selected (valid options: sym,mgi)"

the other "Error: it was not possible to map the input."

And the last "Output output: Test output file (test_output.tab) is missing. If you are using planemo, try adding --update_test_data to generate it."

**Comment by AndreaFurlani** [comment]
Hi @bgruening , now the tests are running correctly. Thank you for your advice.

**Comment by bgruening** [comment]
Wonderful, I have created https://github.com/INFRAFRONTIERDIB/tools-iuc/pull/1/ and added a few more comments there.

**Comment by bgruening** [comment]
2 errors are:

image

test 3 and 5.

and test 4 is:

image

**Comment by AndreaFurlani** [comment]
> 2 errors are:
> 
> image
> 
> test 3 and 5.
> 
> and test 4 is:
> 
> image

I saw the reports, but still I don't get it why the tests are failing. Inputs are correct since by removing the conditionals they all succeed

File: tools/query_impc/impc_tool.xml [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+   <test>
+    <conditional name="inp_q">
+     <param name="selector" value="1"/>
+     <param name="head" value="True"/>
+     <param name="input" value="test_query_1.txt"/>
+     <param name="inp_sel" value="txt"/>
+     <param name="sep" value="t"/>
+    </conditional>
+    <output name="output" file="test_output_1_1.tabular"/>
+   </test>
```
Comment (bgruening) [inline tools/query_impc/impc_tool.xml:195]:
```suggestion
     <test expect_num_outputs="1">
        <conditional name="query_type">
            <param name="selector" value="1"/>
            <conditional name="inp_q">
                <param name="inp_sel" value="txt"/>
                <param name="input" value="test_query_1.txt"/>
                <param name="sep" value="t"/>
            </conditional>
            <param name="head" value="True"/>
        </conditional>
        <output name="output" file="test_output_1_1.tabular"/>
     </test>
```

Comment (AndreaFurlani) [reply tools/query_impc/impc_tool.xml:195]: Thank you @bgruening for the help, now all the tests are working

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks a lot @AndreaFurlani. Great contribution!

If this gets merged it will be automatically deployed to the Galaxy ToolShed. If you would like to see this tool on the European Galaxy Server you can follow step 3 here: https://usegalaxy-eu.github.io/posts/2020/08/22/three-steps-to-galaxify-your-tool/

Thanks again!

**Comment by AndreaFurlani** [comment]
> Thanks a lot @AndreaFurlani. Great contribution!
> 
> If this gets merged it will be automatically deployed to the Galaxy ToolShed. If you would like to see this tool on the European Galaxy Server you can follow step 3 here: https://usegalaxy-eu.github.io/posts/2020/08/22/three-steps-to-galaxify-your-tool/
> 
> Thanks again!

Thank you again @bgruening for all the help. We are going to follow step 3 then to get our tool on the European Galaxy Server.

**Comment by AndreaFurlani** [comment]
> Thanks a lot @AndreaFurlani. Great contribution!
> 
> If this gets merged it will be automatically deployed to the Galaxy ToolShed. If you would like to see this tool on the European Galaxy Server you can follow step 3 here: https://usegalaxy-eu.github.io/posts/2020/08/22/three-steps-to-galaxify-your-tool/
> 
> Thanks again!

Hi @bgruening , regarding step 3 do I need to create a new branch or can I push on this one directly after creating the yaml file? Also it is not clear how should I call it. Thank you for the help

**Pull request merged**